agent/pool/conn.go

```diff
@@ -17,4 +17,14 @@ const (
 	// that is supported and it might be the only one there
 	// ever is.
 	RPCTLSInsecure = 7
+	
+	// NOTE: Currently we use values between 0 and 7 for the different
+	// "protocols" that we may ride over our "rpc" port. I had an idea of
```

Comment:
Maybe remove the first-person pronoun use and phrase it like "We could use TLS + ALPN..."